packages/uniforms/__suites__/connectField.tsx

```diff
@@ -0,0 +1,235 @@
+import { fireEvent, screen } from '@testing-library/react';
+import omit from 'lodash/omit';
+import React from 'react';
+import {
+  BaseForm,
+  Context,
+  OnChange,
+  UnknownObject,
+  connectField,
+  filterDOMProps,
+  randomIds,
+} from 'uniforms';
+
+import { render } from './render';
+
+export function testConnectField() {
+  const onChange = jest.fn();
+
+  const schema = {
+    another: { type: String, optional: true },
+    field: { type: Object, label: 'Field' },
+    'field.subfield': { type: String, label: 'Subfield' },
+  };
+
+  const reactContext = {
+    changed: false,
+    changedMap: {},
+    error: undefined,
+    model: {},
+    name: [],
+    onChange,
+    onSubmit() {},
+    randomId: randomIds(),
+    state: {
+      disabled: false,
+      readOnly: false,
+      showInlineError: true,
+    },
+    submitted: false,
+    submitting: false,
+    validating: false,
+    formRef: {} as BaseForm<UnknownObject>,
+  } as Partial<Context<any>>;
+
+  const Test = (props: UnknownObject & { onChange: OnChange<string> }) => {
+    return props.children ? (
+      <>
+        <input
+          data-testid="field"
+          {...filterDOMProps(omit(props, 'children'))}
+          onChange={event => props.onChange(event.target.value)}
+        />
+        {props.children}
+      </>
+    ) : (
+      <input
+        data-testid="field"
+        {...filterDOMProps(props)}
+        onChange={event => props.onChange(event.target.value)}
+      />
+    );
+  };
+
+  beforeEach(() => {
+    onChange.mockClear();
+  });
+
+  test('connectField - when called, creates component', () => {
+    const Field = connectField(Test);
+
+    expect(Field).toBeInstanceOf(Function);
+    expect(Field.Component).toEqual(Test);
+  });
+
+  test('connectField - when called, does not set default value', () => {
```

Comment:
I'm not sure either about the test name or the test body
Does not set default value or default options? (default value of options maybe?)

I'm asking because the tests below (the ones with the `{ kind: '...' }` are named differently, but the test bodies are similar.